elyra/pipeline/pipeline_definition.py

```diff
@@ -0,0 +1,281 @@
+        return self._node['app_data']['ui_data'].get('label', self._node['app_data']['label'])
+
+    def get_component_parameter(self, key: str, default_value=None) -> Any:
+        self._validate()
+        return self._node['app_data']['component_parameters'].get(key, default_value)
+
+    def set_component_parameter(self, key: str, value: Any):
+        if not key:
+            raise ValueError("Key is required")
+
+        if not value:
+            raise ValueError("Value is required")
+
+        self._node['app_data']['component_parameters'][key] = value
+
+    def validate(self) -> list:
+        validation_issues = list()
+        if "component_parameters" not in self._node['app_data']:
+            validation_issues.append("Node is missing 'component_parameters' field")
+
+        return validation_issues
+
+
+class PipelineDefinition(object):
+    """
+    Represents a helper class to manipulate pipeline json structure
+    """
+    _primary_pipeline = None
+
+    def __init__(self, pipeline_path: Optional[str] = None, pipeline_definition: Optional[Dict] = None):
+        """
+        The constructor enables either passing a pipeline path or the content of the pipeline definition.
+
+        """
+        if not pipeline_path and not pipeline_definition:
+            # at least one parameter should be provided
+            raise ValueError("At least one parameter must be provided ('pipeline_path' or 'pipeline_definition')")
+        if pipeline_path and pipeline_definition:
+            # only one parameter should be provided
+            raise ValueError("Only one parameter should be provided ('pipeline_path' or 'pipeline_definition')")
+
+        if pipeline_path:
+            # supporting loading pipeline from file
+            if not os.path.exists(pipeline_path):
+                raise ValueError(f"Pipeline file not found: '{pipeline_path}'\n")
+
+            with open(pipeline_path) as f:
+                try:
+                    self._pipeline_definition = json.load(f)
+                except ValueError as ve:
+                    raise ValueError(f"Pipeline file is invalid: \n {ve}")
+        else:
+            # supporting passing the pipeline definition directly
+            self._pipeline_definition = pipeline_definition
+
+    @property
+    def id(self) -> str:
+        return self._pipeline_definition.get('id')
+
+    @property
+    def schema_version(self) -> str:
+        return self._pipeline_definition.get('version')
+
+    @property
+    def pipelines(self) -> list:
+        if 'pipelines' not in self._pipeline_definition:
+            raise ValueError("Pipeline is missing 'pipelines' field.")
+        elif len(self._pipeline_definition['pipelines']) == 0:
+            raise ValueError("Pipeline has zero length 'pipelines' field.")
+
+        pipelines: list = list()
+        for pipeline in self._pipeline_definition['pipelines']:
+            pipelines.append(Pipeline(pipeline))
+
+        return pipelines
```

Comment:
Same a `nodes()` - we should consider building this only once.

We could also consider making these (this and `PIpeline.nodes`) coroutines so that they don't require accumulation.